model/test_log.go

```diff
@@ -60,14 +64,31 @@ func FindOneTestLog(name, task string, execution int) (*TestLog, error) {
 	return tl, errors.WithStack(err)
 }
 
+func findAllTestLogs(query db.Q) ([]TestLog, error) {
+	var result []TestLog
+	if err := db.FindAllQ(TestLogCollection, query, &result); err != nil {
+		return nil, errors.Wrap(err, "finding test logs")
+	}
+	return result, nil
+}
+
 func DeleteTestLogsWithLimit(ctx context.Context, env evergreen.Environment, ts time.Time, limit int) (int, error) {
-	if limit > 100*1000 {
-		panic("cannot delete more than 100k documents in a single operation")
+	if limit > maxDeleteCount {
+		return 0, errors.Errorf("cannot delete more than %d documents in a single operation", maxDeleteCount)
+	}
+
+	docsToDelete, err := findAllTestLogs(db.Query(bson.M{TestLogIdKey: bson.M{"$lt": primitive.NewObjectIDFromTimestamp(ts).Hex()}}).WithFields(TestLogIdKey).Limit(limit))
```

Comment:
To confirm: This will read up to 100,000 test docs into memory. That could easily be 10s of megabytes. I don't _know_ if that's a problem.

Reply:
That's a good point. Maybe around [37MB](https://go.dev/play/p/6Kqq-nDFv0G) for the results docs? 
It seems okay? We should only have one of these at a time.